pkg/migration/migration.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+
+	"sigs.k8s.io/krew/pkg/constants"
+	"sigs.k8s.io/krew/pkg/environment"
+	"sigs.k8s.io/krew/pkg/index"
+	"sigs.k8s.io/krew/pkg/migration/oldenvironment"
+)
+
+const (
+	krewPluginName = "krew"
+)
+
+func IsMigrated(newPaths environment.Paths) bool {
+	_, err := os.Lstat(newPaths.InstallReceiptPath())
+	return err == nil
+}
+
+// DoMigration searches for installed plugins, removes each plugin and reinstalls afterwards.
+// Once started, it keeps going even if there are errors.
+func DoMigration(newPaths environment.Paths) error {
+	if IsMigrated(newPaths) {
+		glog.Infoln("Already migrated")
+		return nil
+	}
+
+	oldPaths := oldenvironment.MustGetKrewPaths()
```

Comment:
why do we need `oldenvironment` package?

Can't we just get KREW_ROOT from the current `environment` pkg and do `filepath.Join(root, "store")` etc? This `oldenvironment` adds a lot of code, I recommend just going for straight brain surgery on version==0.2.x assumption here and "assume" paths.

Reply:
My thinking was
- it eases review, because one can do file-diff.
- I don't have to bother about correctness wrt `KREW_ROOT` or so, because it's all tried and tested.
- it raises the abstraction level of the migration code.

Granted, it adds more code than necessary, but it's going to be deleted anyways.

Reply:
I'm OK with it. Don't forget to add TODOs to the methods and packages we must delete. Granted, we'll probably delete this code in v0.4.

Reply:
My hope was that we simply revert the whole commit. I find todos on every method a little excessive (especially since I'm not a big fan of todos). Can we take a middle ground where there's only one todo for a whole package, when the package should be deleted, and one todo for a function outside of such packages?